internal/controllers/core/cluster/cache.go

```diff
@@ -0,0 +1,104 @@
+package cluster
+
+import (
+	"errors"
+	"fmt"
+	"sync"
+	"time"
+
+	"k8s.io/apimachinery/pkg/types"
+
+	"github.com/tilt-dev/tilt/internal/docker"
+	"github.com/tilt-dev/tilt/internal/k8s"
+	"github.com/tilt-dev/tilt/pkg/apis/core/v1alpha1"
+)
+
+// NotFoundError indicates there is no cluster client for the given key.
+var NotFoundError = errors.New("cluster client does not exist")
+
+// ClientCache provides cached clients for use by reconcilers.
+//
+// All clients are goroutine-safe.
+type ClientCache interface {
+	// GetK8sClient returns the Kubernetes client for the cluster or an error for unknown clusters, connections
+	// in a transient error state, or if the connection is of a different type (i.e. Docker).
+	GetK8sClient(clusterKey types.NamespacedName) (k8s.Client, error)
+
+	// GetDockerClient returns the Docker client for the cluster or an error for unknown clusters, connections
```

Comment:
is there a reason we need to expose the `docker.Client`? I'd probably avoid it if we can.

The phrase "the Docker client for the cluster" already means something very different for the rest of the codebase.

You're using it here to mean "the Docker instance that Docker Compose runs its container on."

In other places in the codebase, we use it to mean "the Docker instance that Minikube runs its containers on"

Reply:
I removed it from the interface and renamed the (now-inaccessible) method to `GetComposeDockerClient` in 7a4b44be9, which is awkward but hopefully clear(er) re: purpose.

(We will need it eventually when adding DC support to LU API, no? So I didn't want to totally remove the logic/tests for it. But if I'm misunderstanding and it's totally unnecessary lmk and I'll purge it)